### PR TITLE
Update docs theme to 1.8

### PR DESCRIPTION
**Pull request by dgarcia360**
Related issue https://github.com/scylladb/sphinx-scylladb-theme/issues/1229

ScyllaDB Sphinx Theme 1.8 is out 🥳!  
image

Here's a summary of changes:
- Dark theme support for a fresh new look.
- Updated branding assets including new logos and mascots.
- Improved build process by recommending the inclusion of the `poetry.lock` file to ensure consistent dependencies in your CI/CD pipeline.
- New [tooltip component](https://sphinx-theme.scylladb.com/master/examples/tooltips.html) to add more context to glossary terms, configuration parameters, or any other elements without cluttering the page.

You can read more about all the notable changes [here](https://sphinx-theme.scylladb.com/master/upgrade/CHANGELOG.html#sept-2024).

## How to test this PR

1. Clone this PR.
2. Enter the docs folder, and run:
  
    ```
    make setup
    make preview
    ````

3. Open http://localhost:5500 in your browser. The doc should render without errors, and the version should be Sphinx Theme version (see the footer) must be ``1.8.x``:

    image

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+updates:
+  - package-ecosystem: "pip"
+    directory: "/docs"
+    schedule:
+      interval: "daily"
+    ignore:
+      - dependency-name: "*"
+    allow:
+      - dependency-name: "sphinx-scylladb-theme"
+      - dependency-name: "sphinx-multiversion-scylla"
```
Comment (rzetelskik) [inline .github/dependabot.yml:11]: Iirc we didn't have that good of an experience with depandabot and had to manually fix the PRs either way, so I'm personally not a fan of adding this. Imo it shouldn't be mixed in this PR either way.

**Review by rzetelskik — COMMENTED** [review]
@dgarcia360 thanks for the PR!

The tests are failing with
```
The Poetry configuration is invalid:
  - Additional properties are not allowed ('package-mode' was unexpected)
```
so that's something that needs looking into.

Also please update the commit and PR title to describe the change using the verb in a base form, like "Update docs theme to 1.8". Our contribution guide is outdated - sorry! We're working on updating it just now.

**Comment by rzetelskik** [comment]
we'll need to update our CI first as we're at poetry 1.5

**Comment by rzetelskik** [comment]
Poetry image has been updated.
/test docs docs-multiversion

**Comment by rzetelskik** [comment]
Python was updated to 3.12 in poetry-1.8 image.
/test docs docs-multiversion

File: .github/dependabot.yml (hunk 1 above)
Comment (zimnx) [reply .github/dependabot.yml:11]: +1

**Comment by dgarcia360** [comment]
@rzetelskik deleted dependabot file, we can discuss if we want to include it as part of a separate issue

**Review by zimnx — APPROVED** [review]
lgtm
/assign rzetelskik

**Review by rzetelskik — APPROVED** [review]
/lgtm
Thanks!

**Pull request merged**